### DISCUSSION TITLE
unreachable - need to frag (mtu 1438) with (smart?) linux clients

### DISCUSSION THREAD

**Discussion by f1-outsourcing** [strongswan/strongswan#2158] [Q&A]
I have issues with http traffic (ping works fine) only with linux clients configured via the gnome-gui strongswan client. The http traffic with windows, macos native ipsec clients is ok, strongswan apk on android is also ok.
Even stranger is that I have a host b which is really old (thus really old kernel) which I can access with http traffic. 

So maybe hosts with new kernel 5.x and clients with new kernel get 'confused' in setting up the communication?

These are the messages I see in tcp dump on host A 

 unreachable - need to frag (mtu 1438),
length 590: strongswan.ip > a.ip : ICMP client.ip unreachable - need to frag (mtu 1438), length 556

```
     +--------+      +--------+       +--------+   
     |        |      |        |       |        | 
     |        |      |        |       |        |   
     | client +----->| SSWAN  +------>|   A    |   
     |        |      |        |       |        | 
     +--------+      +--------+       +--------+ 
```

I don't even know where I should change something on host A or on the linux strongswan clients.

**Comment by tobiasbrunner** [comment]
You might want to read [the docs](https://docs.strongswan.org/docs/5.9/howtos/forwarding.html#_mtumss_issues). Why it only affects these particular clients could be because of the negotiated algorithms or some other difference (e.g. what protocol the browser and web server use).

**Reply by f1-outsourcing** [reply]
Yes already saw that page. But I was hoping to narrow down what the issue could be. Because I don't want to break anything that is already working.

**Reply by f1-outsourcing** [reply]
Nothing works on this page

if I add this in charon.conf
```
    plugins {
        kernel-netlink {
                mtu = 1480
                mss = 1481
        }
    }
```

I still have an interface created with

`nm-xfrm-1214752: flags=193<UP,RUNNING,NOARP>  mtu 1400`

This also does not do anything

```
iptables -t mangle -A FORWARD -m policy --pol ipsec --dir in  -p tcp -m tcp --tcp-flags SYN,RST SYN -m tcpmss --mss 1361:1536 -j TCPMSS --set-mss 1360
iptables -t mangle -A FORWARD -m policy --pol ipsec --dir out -p tcp -m tcp --tcp-flags SYN,RST SYN -m tcpmss --mss 1361:1536 -j TCPMSS --set-mss 1360
```

**Reply by f1-outsourcing** [reply]
Is there a command that dumps the active configuration? So can verify settings?

**Reply by tobiasbrunner** [reply]
> This also does not do anything

Did you add these on the client? If so, then yes, this won't do much as most traffic there will not go through FORWARD (instead it goes through INPUT/OUTPUT). These rules are meant to be used on the server.




> I still have an interface created with

What strongSwan version is running on the client? Depending on the version, the interface is not used correctly. On the other hand, with newer versions, there is the `charon-nm.mtu` option that allows reducing the MTU of the device.

> Is there a command that dumps the active configuration? So can verify settings?

What configuration and settings are you referring to? Against what would you compare/verify?

**Comment by f1-outsourcing** [comment]
This route in table 220 is not correctly set, if I do this packets are reduced, and I can access https server.

```
ip route del 192.168.x.0/26 via 192.168.a.1 dev wlo1 proto static src 192.168.x.b table 220
ip route add 192.168.x.0/26 via 192.168.a.1 dev wlo1 proto static src 192.168.x.b advmss 1360 table 220

```

**Reply by tobiasbrunner** [reply]
You have to configure `charon-nm.plugins.kernel-netlink.mss` for this to get applied by charon-nm (the NetworkManager plugin). The `charon` settings don't apply to that daemon.

**Comment by f1-outsourcing** [comment]
Does this not start to look like something default is incorrectly set? I have 2 out of 2 linux clients failing

**Reply by tobiasbrunner** [reply]
It's an MTU issue that depends on a lot of things (like negotiated algorithms, the WAN technology used, the ISP etc.). The default MTU of 1400 for the XFRM interface is already relatively conservative. However, the route in table 220 does not go via XFRM interface in your case, which indicates that you are using a version that doesn't use that interface correctly (5.9.10 or 5.9.11, it was fixed with 5.9.12).

**Reply by f1-outsourcing** [reply]
I did do an upgrade to fedora 39 before perusing this any further. Looking forward to get that version.

**Reply by tobiasbrunner** [reply]
For some reason 5.9.11 is the latest version that's packaged for Fedora (there is apparently [a ticket](https://bugzilla.redhat.com/show_bug.cgi?id=2250666)).
